thirdparty/runtime_environments.go

```diff
@@ -101,20 +100,27 @@ type OSInfo struct {
 	Name    string `json:"name"`
 }
 
+// OSInfoFilterOptions represents the filtering options for GetOSInfo. Each argument is optional except for the AMI field.
+type OSInfoFilterOptions struct {
+	AMI   string
```

Comment:
Does this need JSON tags? Or if it's not meant to be used in a JSON payload and is just used to pass parameters to `GetOSInfo`, adding a `json:"-"` would clarify that.